src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -350,7 +350,7 @@ private bool ShutdownHostIfUnhealthy()
                 // initiating shutdown
                 var message = $"Host unhealthy count exceeds the threshold of {_healthMonitorOptions.Value.HealthCheckThreshold} for time window {_healthMonitorOptions.Value.HealthCheckWindow}. Initiating shutdown.";
                 _logger.LogError(0, message);
-                var environment = _rootServiceProvider.GetService<IScriptJobHostEnvironment>();
+                var environment = _rootServiceProvider.GetRequiredService<IScriptJobHostEnvironment>();
```

Comment:
This is the point in the code where the null ref currently happens, meaning we aren't doing an actual shutdown, but an abnormal termination

Reply:
Can we get this via constructor injection so we don't need to use GetRequiredService?

Reply:
If the change stands, that would be preferable, but I really think the bug is that we were pulling the service from the wrong place. It would have been pulled from the host instead.

Reply:
Yes, for the `IScriptJobHostEnvironment`, that would be my suggestion. It's possible this was using the `IWebScriptJobHostEnvironment` initially, and that's why it was pulling it from the root container.

Reply:
What is IWebScriptJobHostEnvironment? I don't see that type defined anywhere.